scripts/lib-franklin.js

```diff
@@ -377,36 +377,46 @@ export async function getProductDB() {
   return window.productDB;
 }
 
-export async function getProduct(productCode, language) {
-  const productDB = await getProductDB();
+function productDBMatches(entry, country, language) {
```

Comment:
just cosmetic, but i'd rewrite it as:

```
function productDBMatches(entry, country, language) {
   if (country) {
     const countries = entry.Countries.toUpperCase().split('|');
     if (!countries.includes(country.toUpperCase())) {
       return false;
     }
   }

   if (language) {
     const languages = entry.Languages.toUpperCase().split('|');
     if (!languages.includes(language.toUpperCase())) {
       return false;
     }
   }

   return true;
 }
```

- Instead of splitting the countries and languages every time you call the function, consider splitting them beforehand if you are calling this function multiple times.
- If country or language is not provided, it's unnecessary to even perform the checks. Using early returns can reduce unnecessary operations.
- String operations like toUpperCase() are costly. You can minimize the use of these methods.